src/ceph-daemon

```diff
@@ -0,0 +1,1290 @@
+#!/usr/bin/env python3
+
+DEFAULT_IMAGE='ceph/daemon-base'
+DATA_DIR='/var/lib/ceph'
+LOG_DIR='/var/log/ceph'
+UNIT_DIR='/etc/systemd/system'
+VERSION='unknown development version'
+
+PODMAN_PREFERENCE = ['podman', 'docker']  # prefer podman to docker
+
+"""
+You can invoke ceph-daemon in two ways:
+
+1. The normal way, at the command line.
+
+2. By piping the script to the python3 binary.  In this latter case, you should
+   prepend one or more lines to the beginning of the script.
+
+   For arguments,
+
+       injected_argv = [...]
+
+   e.g.,
+
+       injected_argv = ['ls']
+
+   For reading stdin from the '--config-and-json -' argument,
+
+       injected_stdin = '...'
+"""
+
+import argparse
+import configparser
+import json
+import logging
+import os
+import socket
+import subprocess
+import sys
+import tempfile
+import time
+import uuid
+from distutils.spawn import find_executable
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    pass
+try:
+    from io import StringIO
+except ImportError:
+    pass
+
+logging.basicConfig(level=logging.INFO)
+
+podman_path = None
+
+##################################
+
+def pathify(p):
+    if '/' not in p:
+        return './' + p
+    return p
+
+def get_hostname():
+    return socket.gethostname()
+
+def make_fsid():
+    return str(uuid.uuid1())
+
+def is_fsid(s):
+    try:
+        uuid.UUID(s)
+    except ValueError:
+        return False
+    return True
+
+def makedirs(dir):
+    os.makedirs(dir, exist_ok=True)
```

Comment:
should change the mode for these - default is 777, and doesn't apply to intermediate directories - for those it uses umask (https://docs.python.org/3/library/os.html#os.makedirs)

Reply:
see newest patch